include/kodlab_mjbots_sdk/robot_interface.h

```diff
@@ -0,0 +1,143 @@
+         * @param joint_vect a vector of shared pointers to jointbase defining the motors in the robot
+         * @param soft_start_duration how long in dt to spend ramping the torque
+         * @param robot_max_torque the maximum torque to allow per motor in the robot
+         */
+        template <class JointDerived = JointBase>
+        RobotInterface( std::vector<std::shared_ptr<JointDerived>> joint_vect, 
+                        float robot_max_torque,
+                        int soft_start_duration )
+                        : soft_start_(robot_max_torque, soft_start_duration)
+        {
+            // Ensure at compile time that the template is JointBase or a child of JointBase
+            static_assert(std::is_base_of<JointBase, JointDerived>::value);
+
+            // Set up joint vector and joint states
+            for (std::shared_ptr<JointDerived> j : joint_vect){
+                joints.push_back(j); // Copy constructed, uses an implicit upcasting
+                positions_.push_back( j->get_position_reference() );
+                velocities_.push_back( j->get_velocity_reference() );
+                torque_cmd_.push_back( j->get_servo_torque_reference() ); 
+            }
+
+            // Set number of joints
+            num_joints_ = joint_vect.size();
+        }
+
+        /*!
+         * @brief Destroy the Robot Interface object. Virtual destructor for proper derived pointer destruction.
+         */
+        virtual ~RobotInterface() {};
+
+        /*!
+         * @brief Initialize the robot (e.g. communication or states etc.)
+         */
+        virtual void Init(){};
+
+        /*!
+         * @brief Update the robot state and torque command. 
+         */
+        virtual void Update(){cycle_count_++;}; //TODO remove cycle_count and use time or more intelligently set cycle_count or properly handle softstart
+
+        /*!
+         * @brief Stop the robot by setting torques to zero. Can be overridden
+         *        if other features are available e.g. mehanical or regen braking
+         */
+        virtual void Stop(){SetTorques(std::vector<float>(num_joints_, 0));}
+
+        /*!
+         * @brief accessor for joint positions, takes into account direction and offset
+         * @return the joint positions
+         */
+        std::vector<float> GetJointPositions();
+
+        /*!
+         * @brief accessor for joint velocities, takes into account direction
+         * @return the joint velocities
+         */
+        std::vector<float> GetJointVelocities();
+
+        /*!
+         * @brief setter for torque command
+         * @param torques[in] vector of torques
+         */
+        void SetTorques(std::vector<float> torques);
+
+        /*!
+         * @brief accessor for the torque md, takes into account direction
+         * @return the torque cmd
+         */
+        std::vector<float> GetJointTorqueCmd();
+
+        //TODO virtual Attitude GetAttitude()
+
```

Comment:
Create a find joint_id_by_name_fatal function

Reply:
What exactly do you mean by fatal? Should we handle the id error and CTRL-C? Allow the error to kill the program? Or some other jargon I don't know? 
Also, we will be making a map as documented in #28 and this will be handled there instead.